va/va.c

```diff
@@ -36,8 +36,35 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#if defined(_WIN32)
+#include <windows.h>
+#include <io.h>
+typedef HMODULE library_t;
+typedef unsigned int __uid_t;
+inline char* strtok_impl(char *s, const char *delim, char **save_ptr) { return strtok_s(s, delim, save_ptr); }
+inline void* open_library(const char *file, int mode) { return LoadLibrary(file); }
+inline int close_library (void *handle) { return FreeLibrary(handle); }
+inline void *get_symbol_address (void *handle, const char *name) { return GetProcAddress(handle, name); }
+inline __uid_t geteuid() { return 0; };
+inline __uid_t getuid() { return 0; };
+#define ACCESS_OK 0
+#define DRIVER_EXTENSION    ".dll"
+#define DRIVER_PATH_STRING  "%s\\%s%s"
+#define ENV_VAR_SEPARATOR ";"
```

Comment:
Bit of an illustrative example on my earlier suggestion:
- create compat_win32.h or alike
- in there create a win32 "variant" of API respective APIs - `inline void *dlopen(...); inline char *strtok_r(...);....`
- use a `DRIVER_EXTENSION` similar to the *nix one - `#define DRIVER_EXTENSION "_drv_video.dll"

Keep the above a separate preparatory commit.

Reply:
Sounds good, thanks for providing the example. Will look like a less intrusive code change to common code done this way with the win32 compat header. For the DRIVER_EXTENSION, I might be missing out on the historical reasons for the "_drv_video" suffix, but if possible, we'd like to have more flexibility (ie. no name suffix) in the driver name, especially for dynamically loaded driver names (ie. map LUID to driver name from the registry) that could follow different conventions/case naming for each vendor.

Reply:
Many things on *nix follow naming patterns. Don't know if there's a particular reason for "drv_video" over any other, but as you have seen with the mesa vaon12 - the suffix is there. Unless there is a compelling reason, we should not.

Vendors, assuming we get any, still have flexibility - they can name the driver "i-like_my-bikeShed_InPurple_drv_video.dll" if they choose to do so.